### PR TITLE
Fix for reading epoch due to scnlib format changes

### PR DESCRIPTION
**Pull request by GPayne**

### CODE REVIEW

File: modules/space/kepler.cpp [hunk 1]
```diff
@@ -266,15 +266,20 @@ namespace {
             epoch.end(),
             [](char c) { return c == '-'; }
         );
-        const std::string format = (nDashes == 2) ? "{:4}-{:2}-{:2}{}" : "{:4}{:2}{:2}{}";
-        auto res = scn::scan<int, int, int, double>(e, scn::runtime_format(format));
+        const std::string format = (nDashes == 2) ? "{:4d}-{:2d}-{}" : "{:4d}{:2d}{}";
+        auto res = scn::scan<int, int, double>(e, scn::runtime_format(format));
```
Comment (alexanderbock) [inline modules/space/kepler.cpp:270]:
```suggestion
        auto res = scn::scan<int, int, double>(
            e,
            (nDashes == 2) ? "{:4d}-{:2d}-{}" : "{:4d}{:2d}{}"
        );
```

Comment (alexanderbock) [reply modules/space/kepler.cpp:270]: This would skip the extra complexity of the `runtime_format`

File: modules/space/kepler.cpp [hunk 2]
```diff
@@ -266,15 +266,20 @@ namespace {
...
         }
-        auto [year, monthNum, dayOfMonthNum, fractionOfDay] = res->values();
+        auto [year, monthNum, dayOfMonth] = res->values();
         const int daysSince2000 = countDays(year);
-        const int wholeDaysInto = daysIntoGivenYear(year, monthNum, dayOfMonthNum);
-        const double daysInYear = static_cast<double>(wholeDaysInto) + fractionOfDay;
+        const int daysInto = daysIntoGivenYear(
+            year,
+            monthNum,
+            static_cast<int>(dayOfMonth)
```
Comment (alexanderbock) [inline modules/space/kepler.cpp:279]: Shouldn't `dayOfMonth` already be an `int`?

**Review by alexanderbock — APPROVED** [review]

**Comment by alexanderbock** [comment]
To close the issue directly with this PR, you'll need to use the # when referencing the issue number in the title or in the first post

File: modules/space/kepler.cpp (hunk 1 above)
Comment (GPayne) [reply modules/space/kepler.cpp:270]: I tried this, but got error:
`the value of ‘nDashes’ is not usable in a constant expression`
so I'll keep it as-is.

File: modules/space/kepler.cpp (hunk 2 above)
Comment (GPayne) [reply modules/space/kepler.cpp:279]: I changed it from the previous code which read the integer part first, then the decimal value. `dayOfMonth` now reads the full `double` value from the file as the number of days into the month and the remaining fraction of a day.

**Review by Roxeena — COMMENTED** [review]
:+1:

**Pull request merged**